### PR TITLE
Synthetic Monitoring lib

### PR DESCRIPTION
**Pull request by malcolmholmes**
I originally made a PR with this library against Grizzly, but realised it should be here in jsonnet-libs, so moving it over. This lib makes using Synthetic Monitoring much easier, alongside Grizzly.

### CODE REVIEW

File: synthetic-monitoring/sm.libsonnet [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    offset: 0,
+    timeout: 2500,
+    enabled: true,
+    labels: [],
+    target: target,
+    job: name,
+  },
+
+  withHttp():: {
+    settings: {
```
Comment (Duologic) [inline synthetic-monitoring/sm.libsonnet:21]:
```suggestion
    settings+: {
```

Comment (Duologic) [reply synthetic-monitoring/sm.libsonnet:21]: Wouldn't that make more sense?

Comment (malcolmholmes) [reply synthetic-monitoring/sm.libsonnet:21]: Actually, I believe it is one type per check, I should confirm that. In which case `newHttp` etc would make more sense.

Comment (Duologic) [reply synthetic-monitoring/sm.libsonnet:21]: If that is the case, nesting it in `settings.http` looks a bit superfluous.

Comment (malcolmholmes) [reply synthetic-monitoring/sm.libsonnet:21]: exactly.

Comment (malcolmholmes) [reply synthetic-monitoring/sm.libsonnet]: @Duologic I've updated the API to account for this. It is also now slightly more concise. Work for you?

File: synthetic-monitoring/README.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
+# Synthetic Monitoring Jsonnet Library
+
+This library simplifies interaction with the Grafana Synthetic Monitoring API.
```
Comment (Duologic) [inline synthetic-monitoring/README.md:3]: This reads like the library interacts with the API. Can you rephrase? Like 'generates objects/checks for SM' or something along the lines of that.

File: synthetic-monitoring/README.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+Grafana Labs' Synthetic Monitoring is a blackbox monitoring solution provided
+as a part of Grafana Cloud. It provides users with insights into how their
+applications and services are behaving from an external point of view.
+
+It can be configured via an HTTP API. This library allows interaction with
+this API using Jsonnet, making the creation of 'checks' simple.
```
Comment (Duologic) [inline synthetic-monitoring/README.md:10]: Dito, this reads like it actually interacts with the API, but this is not a client.

File: synthetic-monitoring/sm.libsonnet [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+      'Frankfurt',
+      'London',
+      'Paris',
+    ],
+    australasia: [
+      'Sydney',
+    ],
+    continents: ['NewYork', 'Paris', 'Singapore', 'Sydney'],
+    all: self.americas + self.asia + self.europe + self.australasia,
+    small: ['SanFrancisco', 'NewYork', 'Singapore', 'London', 'Sydney'],
```
Comment (Duologic) [inline synthetic-monitoring/sm.libsonnet:101]: 'small' sounds a bit silly, it implies there is also 'big', is this taken from an external example or did you come up with it?

File: synthetic-monitoring/sm.libsonnet [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+      },
+    },
+  },
+
+  tcp: {
+    new(name, target):: _new(name, target) + {
+      settings: {
+        tcp: {
+          ipVersion: 'V4',
+          tlsConfig: {},
```
Comment (Duologic) [inline synthetic-monitoring/sm.libsonnet:33]: Empty object?

**Review by Duologic — APPROVED** [review]
There are a few empty objects/arrays, looks like a documentation example, left a few comments, nothing blocking.

File: synthetic-monitoring/sm.libsonnet (hunk 5 above)
Comment (malcolmholmes) [reply synthetic-monitoring/sm.libsonnet:33]: that's what the API returns for a manually created check, so yes!
